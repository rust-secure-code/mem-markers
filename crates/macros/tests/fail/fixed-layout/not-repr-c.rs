use mem_markers::FixedLayout;

#[derive(FixedLayout)]
struct Foo {
    a: u8,
}

fn main() {}
