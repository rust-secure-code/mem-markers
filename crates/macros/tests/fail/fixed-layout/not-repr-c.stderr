error: FixedLayout trait requires #[repr(C)] or #[repr(transparent)] on structs with fields
 --> $DIR/not-repr-c.rs:3:10
  |
3 | #[derive(FixedLayout)]
  |          ^^^^^^^^^^^
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
